nixos/modules/services/misc/memos.nix

```
@@ -0,0 +1,197 @@
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}:
+let
+  cfg = config.services.memos;
+  envFileFormat = pkgs.formats.keyValue { };
+in
+{
+  options.services.memos = {
+    enable = lib.mkEnableOption "Memos note-taking";
+    package = lib.mkPackageOption pkgs "Memos" {
+      default = "memos";
+    };
+
+    openFirewall = lib.mkEnableOption "opening the ports in the firewall";
+
+    user = lib.mkOption {
+      type = lib.types.str;
+      description = ''
+        The user to run Memos as.
+
+        ::: {.note}
+        If changing the default value, **You** are responsible of creating the corresponding user with {option}`users.users`.
```

Nit: "You" should not be capititalised, it's not a proper noun, nor the start of a new sentence:

```suggestion
        If changing the default value, **you** are responsible of creating the corresponding user with {option}`users.users`.
```

---

Also, IMO it is _usually_ better to use clickable links for option references:

```suggestion
        If changing the default value, **You** are responsible of creating the corresponding user with [](#opt-users.users).
```

Which will link to [users.users](https://nixos.org/manual/nixos/unstable/options#opt-users.users).

IIUC, we _usually_ let the docs renderer auto-fill the link text when referencing options, although I think some people advocate for being explicit so that you get `` `code` ``/`{option}` formatting? \
I.e. ``[{option}`users.users`](#opt-users.users)`` -> [`users.users`](https://nixos.org/manual/nixos/unstable/options#opt-users.users)?

---

(both of the above apply across the diff, of course)